### PR TITLE
ci: run slow unit tests on 3.11 & 3.12

### LINKED ISSUE #259: Update automated testing
Python 3.8 is EOL as of 2024-10-07. We need to update the automated tests and PyPI classifiers of all packages. I'm thinking we should target the three versions that are out of "bugfix" status and 3.12 (the oldest version still in bugfix).

### PR DESCRIPTION
**Pull request by aaraney**
closes #259

### CODE REVIEW

File: .github/workflows/run_slow_unit_tests.yml [hunk 1]
```diff
@@ -87,7 +87,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```
Comment (jarq6c) [inline .github/workflows/run_slow_unit_tests.yml:90]: Good catch :)

File: .github/workflows/run_slow_unit_tests.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
     steps:
     - uses: actions/checkout@v2
```
Comment (jarq6c) [inline .github/workflows/run_slow_unit_tests.yml:43]: I think we'll want to update these, as well. (i.e. `actions/checkout` -> `@v4`)

Comment (aaraney) [reply .github/workflows/run_slow_unit_tests.yml]: Thanks for catching this! Apparently need more coffee this morning 😅

**Review by jarq6c — APPROVED** [review]

**Comment by jarq6c** [comment]
Thanks again @aaraney !

**Pull request merged**